### PR TITLE
Fixed a bug in fake xhr's getResponseHeader function. Plus some compat fixes for th android

### PR DESCRIPTION
**Pull request by mrmaffen**

### CODE REVIEW

File: data/js/tomahawk.js [hunk 1]
```diff
@@ -303,11 +305,12 @@ Tomahawk.Resolver = {
         return RSVP.Promise.resolve(this.testConfig(config)).then(function (results) {
             results = results || Tomahawk.ConfigTestResultType.Success;
             return results;
+        }, function (error) {
+            return error;
```
Comment (dschmidt) [inline data/js/tomahawk.js]: You want to turn the thrown error into a proper return result?

File: data/js/tomahawk.js [hunk 2]
```diff
@@ -49,9 +51,9 @@ RSVP.on('error', function (reason) {
         resolverName = Tomahawk.resolver.instance.settings.name + " - ";
     }
     if (reason) {
-        console.error(resolverName + 'Uncaught error:', reason);
+        Tomahawk.error(resolverName + 'Uncaught error:', reason);
     } else {
-        console.error(resolverName + 'Uncaught error: error thrown from RSVP but it was empty');
+        Tomahawk.error(resolverName + 'Uncaught error: error thrown from RSVP but it was empty');
```
Comment (dschmidt) [inline data/js/tomahawk.js]: Does this keep line numbers in tact? I know they are screwed on desktop anyway, but might get interesting should we ever switch to QWebEngine or maybe you get more meaningful errors in Android already..

File: data/js/tomahawk.js (hunk 1 above)
Comment (mrmaffen) [inline data/js/tomahawk.js]: Yep. Any reasons against it?

File: data/js/tomahawk.js [hunk 3]
```diff
@@ -454,7 +415,7 @@ var doRequest = function(options) {
                 return this.responseHeaders;
             };
             xhr.getResponseHeader = function (header) {
-                return this.responseHeaders[header];
+                return this.responseHeaders[header.toLowerCase()];
```
Comment (dschmidt) [inline data/js/tomahawk.js]: Did you test this on desktop and android?

File: data/js/tomahawk.js (hunk 2 above)
Comment (mrmaffen) [inline data/js/tomahawk.js]: On desktop it simply hands the object over to console.error. So it shouldn't change anything there

File: data/js/tomahawk.js (hunk 3 above)
Comment (mrmaffen) [inline data/js/tomahawk.js]: Correct by spec

File: data/js/tomahawk.js (hunk 1 above)
Comment (dschmidt) [inline data/js/tomahawk.js]: Nope, I figured it was on purpose and I probably know what this is for. Just wanted to make sure this was your intention. I did this several times by accident when I actually meant to rethrow.

**Comment by dschmidt** [comment]
Awesome, looking good.

File: data/js/tomahawk.js (hunk 3 above)
Comment (dschmidt) [inline data/js/tomahawk.js]: Quote?

**Pull request merged**